src/lsptoolshost/dotnetRuntimeExtensionResolver.ts

```diff
@@ -101,4 +123,103 @@ export class DotnetRuntimeExtensionResolver implements IHostExecutableResolver {
 
         return dotnetInfo;
     }
+
+    /**
+     * Checks dotnet --version to see if the value on the path is greater than the minimum required version.
+     * This is adapated from similar O# server logic and should be removed when we have a stable acquisition extension.
+     * @returns true if the dotnet version is greater than the minimum required version, false otherwise.
+     */
+    private async findDotnetFromPath(): Promise<string | undefined> {
+        try {
+            const dotnetInfo = await getDotnetInfo([]);
+            const dotnetVersionStr = dotnetInfo.Version;
+
+            const extensionArchitecture = await this.getArchitectureFromTargetPlatform();
+            const dotnetArchitecture = dotnetInfo.Architecture;
+
+            // If the extension arhcitecture is defined, we check that it matches the dotnet architecture.
+            // If its undefined we likely have a platform neutral server and assume it can run on any architecture.
+            if (extensionArchitecture && extensionArchitecture !== dotnetArchitecture) {
+                throw new Error(
+                    `The architecture of the .NET runtime (${dotnetArchitecture}) does not match the architecture of the extension (${extensionArchitecture}).`
+                );
+            }
+
+            const dotnetVersion = semver.parse(dotnetVersionStr);
+            if (!dotnetVersion) {
+                throw new Error(`Unknown result output from 'dotnet --version'. Received ${dotnetVersionStr}`);
+            }
+
+            if (semver.lt(dotnetVersion, this.minimumDotnetVersion)) {
+                throw new Error(
+                    `Found dotnet version ${dotnetVersion}. Minimum required version is ${this.minimumDotnetVersion}.`
+                );
+            }
+
+            // Find the location of the dotnet on path.
+            const command = this.platformInfo.isWindows() ? 'where' : 'which';
+            const whereOutput = await promisify(exec)(`${command} dotnet`);
+            if (!whereOutput.stdout) {
+                throw new Error(`Unable to find dotnet from ${command}.`);
+            }
+
+            const path = whereOutput.stdout.trim();
+            if (!existsSync(path)) {
+                throw new Error(`dotnet path does not exist: ${path}`);
+            }
```

Comment:
I reused this code for C# Dev Kit, but it failed in AzP because `where` returns one line for each dotnet found on the PATH, and this code assumes exactly one line will be produced. I'll fix my copy. But I suspect your code here will fail on any customer machine that has dotnet.exe installed in two places.

Reply:
Ah - thanks for the info.  I'll take a look.